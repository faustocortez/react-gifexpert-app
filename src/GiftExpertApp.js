import React, { Fragment, useState } from "react";
import { AddCategory } from './components/AddCategory'
import { GifGrid } from "./components/GifGrid";


const GiftExpertApp = ({ defaultCategories = [] }) => {
  // const [categories, setCategories] = useState(['Attack On Titan']);0
  const [categories, setCategories] = useState(defaultCategories);
  
  // const handleAdd = (name) => {
  //   if (categories.indexOf(name) === -1) {
  //     setCategories([...categories, name]);
  //   }
  // };

  return (
    <Fragment>
      <h2>Gift Expert App</h2>
      <AddCategory setCategories={ setCategories }/>

      <hr />

      <ol>
        { categories.map((category) => (
          <GifGrid
            key={ category }
            category={ category }
          />
        )) }
      </ol>
    </Fragment>
  )
}

export default GiftExpertApp;