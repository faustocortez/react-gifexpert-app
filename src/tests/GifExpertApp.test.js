import React from "react";
import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import GiftExpertApp from "../GiftExpertApp";


describe('Pruebas al componente GifExpert App', () => {
    
    test('* debe renderizar el componente de manera correcta', () => {
        
        const wrapper = shallow(<GiftExpertApp />);

        expect(wrapper).toMatchSnapshot();

    });

    test('* debe de mostrar una lista de categorias', () => {
        const categories = ['Nanatsu no taiza', 'Parasyte'];
        const wrapper = shallow(<GiftExpertApp defaultCategories={categories} />);

        expect(wrapper).toMatchSnapshot();
        
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
    
    
});
