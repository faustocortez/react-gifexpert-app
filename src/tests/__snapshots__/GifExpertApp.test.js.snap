// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente GifExpert App * debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    Gift Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Nanatsu no taiza"
      key="Nanatsu no taiza"
    />
    <GifGrid
      category="Parasyte"
      key="Parasyte"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas al componente GifExpert App * debe renderizar el componente de manera correcta 1`] = `
<Fragment>
  <h2>
    Gift Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
